include/ck_tile/ops/epilogue/cshuffle_epilogue.hpp

```diff
@@ -1,194 +1,199 @@
 // SPDX-License-Identifier: MIT
-// Copyright (c) 2018-2024, Advanced Micro Devices, Inc. All rights reserved.
+// Copyright (c) 2018-2025, Advanced Micro Devices, Inc. All rights reserved.
 
 #pragma once
 
 #include "ck_tile/core.hpp"
-
-#define CK_TILE_MAX_RANK 5
+#include "ck_tile/ops/gemm/warp/warp_gemm_dispatcher.hpp"
 
 namespace ck_tile {
 
-// this epilogue aiming to store a matrix with different layout from the shared memory to the global
-// memory.
 template <typename AccDataType_,
           typename ODataType_,
-          bool kPadM_,
-          bool kPadN_,
-          bool kTilePermute_,
-          index_t kRank_,
-          index_t kPerm0,
-          index_t kPerm1,
-          index_t TileSize0,
-          index_t TileSize1,
-          index_t kPerm2    = 0,
-          index_t kPerm3    = 0,
-          index_t kPerm4    = 0,
-          index_t TileSize2 = 0,
-          index_t TileSize3 = 0,
-          index_t TileSize4 = 0>
+          index_t kBlockSize_,
+          index_t kM_,
+          index_t kN_,
+          index_t kMWave_,
+          index_t kNWave_,
+          index_t kMPerXdl_,
+          index_t kNPerXdl_,
+          index_t kKPerXdl_,
+          bool isMFMALayoutTransposed_>
 struct CShuffleEpilogueProblem
 {
-    using AccDataType                                = remove_cvref_t<AccDataType_>;
-    using ODataType                                  = remove_cvref_t<ODataType_>;
-    static constexpr bool kPadM                      = kPadM_;
-    static constexpr bool kPadN                      = kPadN_;
-    static constexpr bool kTilePermute               = kTilePermute_;
-    static constexpr index_t kRank                   = kRank_;
-    static constexpr index_t kPerm[CK_TILE_MAX_RANK] = {kPerm0, kPerm1, kPerm2, kPerm3, kPerm4};
-    static constexpr index_t tile_sizes[CK_TILE_MAX_RANK] = {
-        TileSize0, TileSize1, TileSize2, TileSize3, TileSize4};
+    using AccDataType                               = remove_cvref_t<AccDataType_>;
+    using ODataType                                 = remove_cvref_t<ODataType_>;
+    static constexpr index_t kBlockSize             = kBlockSize_;
+    static constexpr index_t kMPerBlock             = kM_;
+    static constexpr index_t kNPerBlock             = kN_;
+    static constexpr index_t kMWave                 = kMWave_;
+    static constexpr index_t kNWave                 = kNWave_;
+    static constexpr index_t kMPerXdl               = kMPerXdl_;
+    static constexpr index_t kNPerXdl               = kNPerXdl_;
+    static constexpr index_t kKPerXdl               = kKPerXdl_;
+    static constexpr index_t isMFMALayoutTransposed = isMFMALayoutTransposed_;
```

Comment:
In future we may have other instructions than MFMA.
```suggestion
    static constexpr index_t isCTransposed = isCTransposed_;
```